squiggy/models/user.py

```diff
@@ -162,10 +162,18 @@ def find_by_course_id(cls, canvas_user_id, course_id):
         return cls.query.filter(where_clause).one_or_none()
 
     @classmethod
-    def get_users_by_course_id(cls, course_id):
-        return cls.query.filter(
+    def get_users_by_course_id(cls, course_id, sections=None):
+        query = cls.query.filter(
             and_(cls.course_id == course_id, cls.canvas_enrollment_state.in_(['active', 'invited'])),
-        ).order_by(cls.canvas_full_name).all()
+        )
+        if sections:
+            query = query.filter(
+                or_(
+                    cls.canvas_course_sections.overlap(sections),
+                    and_(cls.canvas_course_role.not_ilike('%student%'), cls.canvas_course_role.not_ilike('%learner%')),
+                ),
```

Comment:
Thanks for the extra work to keep in in the instructors, @lyttam - I hadn't thought through that of course they need to stay visible.